core/tests/org.fusesource.ide.camel.model.service.core.tests.integration/src/main/java/org/fusesource/ide/camel/model/service/core/tests/integration/core/io/FuseProject.java

```diff
@@ -144,12 +148,35 @@ public void after() {
 		if (project != null && project.exists()) {
 			try {
 				project.delete(true, new NullProgressMonitor());
-			} catch (CoreException e) {
+				awaitProjectIsRemovedFrom2eRegistry();
+			} catch (CoreException | InterruptedException e) {
 				CamelModelServiceIntegrationTestActivator.getDefault().getLog().log(new Status(IStatus.ERROR, CamelModelServiceIntegrationTestActivator.ID, "Cannot delete project used during test", e));
 			}
 		}
 	}
 
+	/**
+	 * The m2e registry is updated asynchronously.
+	 * When deleting a project, and recreating one with same name just after, it can cause StaleMutableRegistryException.
+	 * Awaiting that the m2e registry is updated with project removal avoids this issue.
+	 * 
+	 * @throws InterruptedException
+	 */
+	private void awaitProjectIsRemovedFrom2eRegistry() throws InterruptedException {
```

Comment:
do u want to rename it to "awaitProjectIsRemovedFromM2ERegistry" or was that missing m intentional?